### PR TITLE
Antagonist Panel for Slaughter Demons, Laughter Demons, and Imps(?!)

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

uses the power of ANIMATION (the flavor text is altered a bit on this for better wording, by the way)

https://user-images.githubusercontent.com/40974010/131225997-c0811cc6-ca21-4b7a-958e-9cf8e8ed503a.mp4

## Why It's Good For The Game

more antag panels means less garbage sent to your precious chat

## Changelog
:cl:
qol: Converted slaughter demons (and their friends) to antagonist panels
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/styles/interfaces/HellishRunes.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@use '../colors.scss';
+@use '../base.scss';
+
+$bg-map: colors.$bg-map !default;
+
+.demon__demonrune {
```
Comment (stylemistake) [inline tgui/packages/tgui/styles/interfaces/HellishRunes.scss:6]: Wouldn't it be better to call this class the same way as the base filename? Or even further, the same way as the interface you coded this for.

Comment (tralezab) [reply tgui/packages/tgui/styles/interfaces/HellishRunes.scss:6]: Aha, so like `HellishRunes__demonrune?`

**Comment by Mothblocks** [comment]
i was expecting cool sygils not sans-serif latin letters

**Comment by Arkatos1** [comment]
Atleast it's not in the Toon Town font

**Comment by tralezab** [comment]
I really tried my best to add a font folks, turns out it's a huge headache so you will take your default dan letters and you will like it

File: code/modules/antagonists/slaughter/imp_antag.dm [hunk 2]
```diff
@@ -18,3 +19,10 @@
 	newobjective.explanation_text = "Try to get a promotion to a higher devilish rank."
 	newobjective.owner = owner
 	objectives += newobjective
+
+/datum/antagonist/slaughter/ui_static_data(mob/user)
```
Comment (Fikou) [inline code/modules/antagonists/slaughter/imp_antag.dm:23]: this isnt a slaughter demon

Comment (tralezab) [reply code/modules/antagonists/slaughter/imp_antag.dm:23]: good point

Comment (Mothblocks) [reply code/modules/antagonists/slaughter/imp_antag.dm]: were imps not removed when we removed devil stuff?

File: tgui/packages/tgui/interfaces/AntagInfoDemon.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      height={356}
+      theme="syndicate">
+      <Window.Content
+        style={{ 'background-image': 'none' }}>
+        <Stack fill>
+          <Stack.Item>
+            <DemonRunes />
+          </Stack.Item>
+          <Stack.Item grow>
+            <Stack vertical width="544px" fill>
```
Comment (Watermelon914) [inline tgui/packages/tgui/interfaces/AntagInfoDemon.tsx:48]: why hardcode the width here? What if someone decides to resize the UI?

File: code/modules/antagonists/slaughter/imp_antag.dm [hunk 4]
```diff
@@ -18,3 +19,10 @@
 	newobjective.explanation_text = "Try to get a promotion to a higher devilish rank."
 	newobjective.owner = owner
 	objectives += newobjective
+
+/datum/antagonist/imp/ui_static_data(mob/user)
+	. = ..()
+	var/list/data = list()
```
Comment (Watermelon914) [inline code/modules/antagonists/slaughter/imp_antag.dm:25]: setting the return value here only to explicitly return at the end of the statement is a bit redundant. Could you change that?

File: code/modules/antagonists/slaughter/slaughter_antag.dm [hunk 5]
```diff
@@ -28,6 +30,15 @@
 	new_objective2.explanation_text = "[objective_verb] everyone[summoner ? " else while you're at it":""]."
 	objectives += new_objective2
 
+/datum/antagonist/slaughter/ui_static_data(mob/user)
+	. = ..()
+	var/list/data = list()
```
Comment (Watermelon914) [inline code/modules/antagonists/slaughter/slaughter_antag.dm:35]: same thing as above here.

File: tgui/packages/tgui/interfaces/AntagInfoDemon.tsx (hunk 3 above)
Comment (tralezab) [reply tgui/packages/tgui/interfaces/AntagInfoDemon.tsx:48]: It gets farked. It's sadly an IE bug i'm pretty sure that forced me to do this

**Comment by Watermelon914** [comment]
you gonna get this done?

**Comment by tralezab** [comment]
done!!!!

**Pull request merged**